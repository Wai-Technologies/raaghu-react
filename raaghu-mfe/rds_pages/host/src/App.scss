//@import "raaghu-react-themes/build/styles/default.css";
// Please dont remove below comment line Adding raaghu-theme in local side.
@import "../../../../raaghu-react-themes/src/styles/default.scss";

.main-body {
    margin-top: 29px;
    position: relative;
}
