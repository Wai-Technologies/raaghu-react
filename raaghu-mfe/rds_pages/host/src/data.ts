import { useTranslation } from "react-i18next";

export const getTranslatedThemeItems = () => {
  const { t } = useTranslation();

  const themeItems = [
    {
      id: 0,
      label: t("Light"),
      val: "Light",
      iconWidth: "30px",
      iconHeight: "30px",
      iconPath: "/assets/lottie-files/outlined/dual-color/sun.json",
    },
    {
      id: 1,
      label: t("Dark"),
      val: "Dark",
      iconWidth: "30px",
      iconHeight: "30px",
      iconPath: "/assets/lottie-files/outlined/dual-color/moon.json",
    },
    {
      id: 2,
      label: t("Semi Dark"),
      val: "Semi Dark",
      iconWidth: "30px",
      iconHeight: "30px",
      iconPath: "/assets/lottie-files/outlined/dual-color/semi-dark.json",
    },
  ];

  return themeItems;
};

export const formsChildren = [
  {
    key: "5-0",
    label: "View",
    path: "/forms-view/:id",
    permission: "Forms.Form",
  },
];

export const getLabelForPath: any = (path: string, navItems: any) => {
  let label = null;
  for (const navItem of navItems) {
    if (navItem.path === path) {
      return navItem.label;
    }
    if (navItem.children) {
      label = getLabelForPath(path, navItem.children);
      if (label) {
        return label;
      }
    }
  }
  return label;
};