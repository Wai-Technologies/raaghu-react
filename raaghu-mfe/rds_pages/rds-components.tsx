export * from "../../raaghu-components/src/index";
