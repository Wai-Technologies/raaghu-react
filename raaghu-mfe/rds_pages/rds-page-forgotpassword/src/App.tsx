import { Suspense } from "react";
import React from "react";
import ForgotPassword from "./forgotpassword/forgotpassword";

const App = () => (
    <Suspense>
        <ForgotPassword></ForgotPassword>
    </Suspense>
);

export default App;
