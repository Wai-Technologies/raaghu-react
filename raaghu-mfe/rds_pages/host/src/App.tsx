import React, { Suspense } from "react";
import "bootstrap/dist/js/bootstrap.bundle.min.js";
import "./App.scss";
import AuthHigherComponent from "./AuthHigherComponent";

const App = () => {
    return (
        <Suspense>
            <AuthHigherComponent></AuthHigherComponent>
        </Suspense>
    );
};

export default App;
