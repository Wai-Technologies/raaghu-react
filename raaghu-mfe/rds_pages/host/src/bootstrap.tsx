import App from "./App";
import ReactDOM from "react-dom/client";
import React from "react";
import { BrowserRouter } from "react-router-dom";
import { store } from "../../../libs/public.api";
import { Provider } from "react-redux";

(ReactDOM as any).createRoot(document.getElementById("root")).render(
    <Provider store={store}>
        <div>
            <BrowserRouter>
                <App />
            </BrowserRouter>
        </div>
    </Provider>
);

