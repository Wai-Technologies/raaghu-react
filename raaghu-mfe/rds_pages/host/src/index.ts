import("./bootstrap");
import "../../../web.config";
