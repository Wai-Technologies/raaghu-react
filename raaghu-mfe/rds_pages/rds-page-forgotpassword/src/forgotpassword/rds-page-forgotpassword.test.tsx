import React from "react";
import "@testing-library/jest-dom";
import { screen, render } from "@testing-library/react";
import ForgotPassword from "./forgotpassword";
import { store } from "../../../../libs/state-management";
import { Provider } from "react-redux";
import { RdsCompForgotPassword } from "../../../rds-components";
import { RdsLabel } from "../../../../../raaghu-elements/src";

jest.mock("react-i18next", () => ({
    useTranslation: () => ({
      t: (key: string) => key,
    }),
    initReactI18next: { // add this line
      type: "3rdParty",
      init: () => {},
    },
  }));
jest.mock("react-router-dom", () => ({
    useNavigate: () => jest.fn(),
}));

global.location = {
    pathname: "../../../../libs/proxy/core",
  } as any;
  
  jest.mock("react-redux", () => ({
    ...jest.requireActual("react-redux"),
    useSelector: jest.fn(),
    useDispatch: jest.fn(),
  }));
  jest.mock("../../../../../raaghu-components/src/index.ts", () => ({
    ...jest.requireActual("../../../../../raaghu-components/src/index.ts"),
  }));
  jest.mock("../../../../../raaghu-elements/src/index.ts", () => ({
    ...jest.requireActual("../../../../../raaghu-elements/src/index.ts"),
  }));
  jest.mock("../../../../libs/state-management/index.ts", () => ({
    store: {
      getState: jest.fn().mockReturnValue({
        persistedReducer: {
          configureStore: {
            configuration: {
              auth: {
                grantedPolicies: {},
              },
            },
          },
        },
      }),
      dispatch: jest.fn(),
      subscribe: jest.fn(),
    },
  }));
  
  // Mock the rds-elements and rds-components
  jest.mock("../../../../../raaghu-elements/src", () => ({
    RdsButton: jest.fn(),
    RdsOffcanvas: jest.fn(),
    RdsAlert: jest.fn(),
    RdsSearch: jest.fn(),
    RdsModal: jest.fn(),
    RdsLabel : jest.fn(),
  }));
  
  jest.mock("../../../../../raaghu-components/src", () => ({
    RdsCompApiScopeBasicResource: jest.fn(),
    RdsCompDatatable: jest.fn(),
    RdsCompAlertPopup: jest.fn(),
    RdsOffcanvas:jest.fn(),
    RdsCompForgotPassword: jest.fn(),
  }));
  
  jest.mock("react-redux", () => ({
    ...jest.requireActual("react-redux"),
    useDispatch: jest.fn(),
  }));
  
  // Mock localStorage in your test setup
  const localStorageMock: Storage = {
    getItem: jest.fn(),
    setItem: jest.fn(),
    removeItem: jest.fn(),
    clear: jest.fn(),
    length: 0, // Add length property
    key: jest.fn(), // Add key method
  };


  global.localStorage = localStorageMock;
  describe("Forgot password Page", () => {
it("renders Forgot Password page correctly", () => {
    render(
        <Provider store={store}>
            <ForgotPassword />
        </Provider>
    );
});

it("Check Forgot Password functionality", () => {
  const forgotPasswordHandler = jest.fn();
  const resendHandler = jest.fn();
  const onLoginHandler = jest.fn();
  const onClickHandler = jest.fn();
  const _currentLanguageLabel = "English";

  const password = render(
      <Provider store={store}>
          <ForgotPassword />
          <RdsCompForgotPassword languageLabel={_currentLanguageLabel} languageData={undefined} onForgotPassword={forgotPasswordHandler} onResend={resendHandler} onLogin={onLoginHandler} onClickHandler={onClickHandler} />
          <RdsLabel label="Â©2023 WAi Technologies. All rights reserved"/>
      </Provider>
  );
  expect(password).toMatchSnapshot();
});

afterEach(() => {
    jest.clearAllMocks(); // Clear mock calls between tests
    // Reset localStorage to a new object
    global.localStorage = {
      getItem: jest.fn(),
      setItem: jest.fn(),
      removeItem: jest.fn(),
      clear: jest.fn(),
      length: 0,
      key: jest.fn(),
    };
  });
  })