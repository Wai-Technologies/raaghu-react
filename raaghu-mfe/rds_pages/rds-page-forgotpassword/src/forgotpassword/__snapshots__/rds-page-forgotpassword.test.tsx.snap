// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Forgot password Page Check Forgot Password functionality 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="align-items-center d-flex justify-content-center m-auto login vh-100"
      >
        <div
          class="container-fluid"
        >
          <div
            class="bg-login align-items-center h-100 row rounded-3 overflow-x-hidden thin-scroll overflow-y-scroll"
          >
            <div
              class="col-xxl-6 col-xl-6 col-lg-12 col-12"
            >
              <div
                class="mx-auto pb-3 pt-3 px-3 px-lg-5 px-md-5 px-xl-5 px-xxl-5 rounded-4 shadow-md-lg w-75 w-xs-100"
              >
                <div>
                  <div
                    class="login-border-bottom mb-4 pb-1 raaghu-logo"
                  >
                    <div
                      class="text-center"
                    >
                      <img
                        height="68px"
                        src="./assets/raaghu_text_logo.svg"
                        width="161px"
                      />
                    </div>
                  </div>
                </div>
                <div
                  class="d-flex justify-content-center pt-2 pt-lg-3 pt-xl-3 pt-xxl-3"
                />
              </div>
            </div>
            <div
              class="col-xxl-6 col-xl-6 col-12 order-1 h-100 order-sm-2 rounded-end position-relative align-items-center p-0 login-card-height d-xxl-block d-xl-block d-lg-none d-none"
              style="background-image: url(../assets/bg_1.png); background-size: cover; background-position: bottom; background-repeat: no-repeat;"
            >
              <video
                autoplay=""
                class="video"
                id="myVideo"
                loop=""
              >
                <source
                  src="../assets/Comp1.mp4"
                  type="video/mp4"
                />
              </video>
              <div
                class="raghu1"
              >
                <img
                  src="../assets/fg_raaghu.png"
                />
              </div>
              <div
                class="wrap"
              >
                <div
                  class="c"
                />
                <div
                  class="c"
                />
                <div
                  class="c"
                />
                <div
                  class="c"
                />
                <div
                  class="c"
                />
                <div
                  class="c"
                />
                <div
                  class="c"
                />
                <div
                  class="c"
                />
                <div
                  class="c"
                />
                <div
                  class="c"
                />
                <div
                  class="c"
                />
                <div
                  class="c"
                />
                <div
                  class="c"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="align-items-center d-flex justify-content-center m-auto login vh-100"
    >
      <div
        class="container-fluid"
      >
        <div
          class="bg-login align-items-center h-100 row rounded-3 overflow-x-hidden thin-scroll overflow-y-scroll"
        >
          <div
            class="col-xxl-6 col-xl-6 col-lg-12 col-12"
          >
            <div
              class="mx-auto pb-3 pt-3 px-3 px-lg-5 px-md-5 px-xl-5 px-xxl-5 rounded-4 shadow-md-lg w-75 w-xs-100"
            >
              <div>
                <div
                  class="login-border-bottom mb-4 pb-1 raaghu-logo"
                >
                  <div
                    class="text-center"
                  >
                    <img
                      height="68px"
                      src="./assets/raaghu_text_logo.svg"
                      width="161px"
                    />
                  </div>
                </div>
              </div>
              <div
                class="d-flex justify-content-center pt-2 pt-lg-3 pt-xl-3 pt-xxl-3"
              />
            </div>
          </div>
          <div
            class="col-xxl-6 col-xl-6 col-12 order-1 h-100 order-sm-2 rounded-end position-relative align-items-center p-0 login-card-height d-xxl-block d-xl-block d-lg-none d-none"
            style="background-image: url(../assets/bg_1.png); background-size: cover; background-position: bottom; background-repeat: no-repeat;"
          >
            <video
              autoplay=""
              class="video"
              id="myVideo"
              loop=""
            >
              <source
                src="../assets/Comp1.mp4"
                type="video/mp4"
              />
            </video>
            <div
              class="raghu1"
            >
              <img
                src="../assets/fg_raaghu.png"
              />
            </div>
            <div
              class="wrap"
            >
              <div
                class="c"
              />
              <div
                class="c"
              />
              <div
                class="c"
              />
              <div
                class="c"
              />
              <div
                class="c"
              />
              <div
                class="c"
              />
              <div
                class="c"
              />
              <div
                class="c"
              />
              <div
                class="c"
              />
              <div
                class="c"
              />
              <div
                class="c"
              />
              <div
                class="c"
              />
              <div
                class="c"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
